src/transformState.js

```diff
@@ -5,7 +5,29 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const toRemove = [];
+
+  for (const elemment of actions) {
+    switch (elemment['type']) {
+      case 'addProperties':
+        Object.assign(state, elemment['extraData']);
+        break;
+      case 'removeProperties':
+        Object.assign(toRemove, elemment['keysToRemove']);
```

Comment:
It's very hard to read solution and it's also bad in terms of performance. Delete required props right away.

Reply:
I'm not sure what you mean here?